support/BRFileService.c

```diff
@@ -265,104 +433,142 @@ fileServiceLoad (BRFileService fs,
-                return fileServiceFailedUnix (fs, buffer, file, errno);
-
-            // read the header
-            switch (headerVersion) {
-                case HEADER_FORMAT_1: {
-                    // read the version
-                    if (1 != fread (&version, sizeof(BRFileServiceVersion), 1, file) ||
-                        // read the checksum
-                        // read the bytesCount
-                        1 != fread (&bytesCount, sizeof(uint32_t), 1, file))
-                        return fileServiceFailedUnix (fs, buffer, file, errno);
-
-                    break;
-                }
-            }
-
-            // Ensure `buffer` is large enough
-            if (bytesCount > bufferSize) {
-                bufferSize = bytesCount;
-
-                uint8_t *bufferNew = realloc (buffer, bufferSize);
-                if (NULL == bufferNew) return fileServiceFailedUnix (fs, buffer, NULL, ENOMEM);
-                buffer = bufferNew;
-            }
-
-            // read the bytes - multiple might be required
-            if (bytesCount != fread (buffer, 1, bytesCount, file))
-                return fileServiceFailedUnix (fs, buffer, file, errno);
-
-            // All file reading is complete; next read should be EOF.
-
-            // Done with file.
-            if (0 != fclose (file))
-                return fileServiceFailedUnix (fs, buffer, NULL, errno);
-
-            // We now have everything
-
-            // This will need some later rework.  If a header includes some data, like a checksum,
-            // we won't have that value in this context when needed.
-
-            // Do something header specific
-            switch (headerVersion) {
-                case HEADER_FORMAT_1:
-                    // compute the checksum
-                    // compare the checksum
-                   break;
-            }
-
-            // Look up the entity handler
-            BRFileServiceEntityHandler *handler = fileServiceEntityTypeLookupHandler(entityType, version);
-            if (NULL == handler) return fileServiceFailedImpl (fs,  buffer, NULL, "missed type handler");
-
-            // Read the entity from buffer and add to results.
-            void *entity = handler->reader (handler->context, fs, buffer, bytesCount);
-            if (NULL == entity) return fileServiceFailedEntity (fs, buffer, NULL, type, "reader");
-
-            // Update restuls with the newly restored entity
-            BRSetAdd (results, entity);
-
-            // If the read version is not the current version, update
-            if (updateVersion &&
-                (version != entityType->currentVersion ||
-                 headerVersion != currentHeaderFormatVersion))
-                fileServiceSave (fs, type, entity);
+    while (SQLITE_ROW == sqlite3_step(fs->sdbSelectAllStmt)) {
+        const char *hash = (const char *) sqlite3_column_text (fs->sdbSelectAllStmt, 0);
```

Comment:
Likely want to defensively treat all input from the FS (i.e. length checks, overflow checks, etc?).

Reply:
Sorry, missing the issue here.

Reply:
In terms of trusting the input values that we are parsing (potentially with just asserts).

Reply:
Sorry, still not sure of what 'input values'. `fs->sdbXYZStmt` or `hash`?  [I need to look at the whole code block?]

Reply:
Hash and data. Are they 0 bytes, is dataBytes big enough to read from offset, as it grows, etc.